plugins/modules/win_dns_client.ps1

```diff
@@ -379,6 +423,25 @@ Try {
         }
     }
 
+}
+Catch {
+    $excep = $_
+
+    Write-DebugLog "Exception: $($excep | Out-String)"
+
+    Throw
+}
+
+Try {
+    if (-not (Test-DnsSuffixSearchList $suffix_search_list)) {
```

Comment:
Might be best to move this whole try block into the above one so we aren't repeating the catch multiple times.